### PR TITLE
NIAD-1843: Application DB setup

### PR DESCRIPTION
**Pull request by mrogowska**

### CODE REVIEW

File: db-connector/config/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,165 @@
+<?xml version="1.0"?>
```
Comment (dawidskainos) [inline db-connector/config/checkstyle/checkstyle.xml:1]: would it be possible to have common config for all modules?

File: db-connector/src/main/java/uk/nhs/adaptors/connector/dao/PatientMigrationRequestDao.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package uk.nhs.adaptors.connector.dao;
+
+import org.jdbi.v3.sqlobject.customizer.Bind;
+import org.jdbi.v3.sqlobject.statement.SqlUpdate;
+
+public interface PatientMigrationRequestDao {
+    @SqlUpdate("INSERT INTO public.patient_migration_request(id) VALUES (:id)")
```
Comment (dawidskainos) [inline db-connector/src/main/java/uk/nhs/adaptors/connector/dao/PatientMigrationRequestDao.java:7]: Perhaps we could remove this one? I don't want anyone to get confused and think we're going to set IDs manually. Those should be set using sequence.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+version: '3.8'
+services:
+    pss_db:
+        build: ./db
+        environment:
+            POSTGRES_HOST_AUTH_METHOD: "trust"
```
Comment (dawidskainos) [inline docker-compose.yml:6]: maybe we could use username/password combination?

File: db/setup-db.sql [hunk 4]
```diff
@@ -0,0 +1,3 @@
+CREATE USER pss;
```
Comment (dawidskainos) [inline db/setup-db.sql:1]: Maybe we could create separate users for gpc-facade and gp2gp translator?

File: gpc-api-facade/src/main/resources/db/changelog/migration/01-create-patient_migration_request.sql [hunk 5]
```diff
@@ -0,0 +1,4 @@
+CREATE TABLE patient_migration_request
+(
+    id character varying(36) NOT NULL
```
Comment (dawidskainos) [inline gpc-api-facade/src/main/resources/db/changelog/migration/01-create-patient_migration_request.sql:3]: 2 comments here:
- maybe we should use liquibase syntax - to abstract from sepecific database?
- can we create sequence for ID column? and actually make it PK?

File: db/setup-db.sql [hunk 6]
```diff
@@ -0,0 +1,3 @@
...
+CREATE DATABASE pss WITH OWNER pss;
+GRANT ALL PRIVILEGES ON DATABASE pss TO pss;
```
Comment (dawidskainos) [inline db/setup-db.sql:3]: can we GRANT specific privileges, not ALL?

File: db-connector/src/main/java/uk/nhs/adaptors/connector/configuration/BaseConfiguration.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+@Configuration
+@EntityScan(basePackages = {"uk.nhs.adaptors.connector"})
+@ComponentScan(basePackages = {"uk.nhs.adaptors.connector"})
+public class BaseConfiguration {
+
+    @Bean
+    public Jdbi jdbi(DataSource ds, List<JdbiPlugin> jdbiPlugins, List<RowMapper<?>> rowMappers) {
+        TransactionAwareDataSourceProxy proxy = new TransactionAwareDataSourceProxy(ds);
+        Jdbi jdbi = Jdbi.create(proxy);
+        jdbiPlugins.forEach(jdbi::installPlugin);
```
Comment (dawidskainos) [inline db-connector/src/main/java/uk/nhs/adaptors/connector/configuration/BaseConfiguration.java:28]: should we add Postgres JDBI Plugin? 
https://jdbi.org/apidocs/org/jdbi/v3/postgres/PostgresPlugin.html

File: db/setup-db.sql [hunk 8]
```diff
@@ -0,0 +1,3 @@
+CREATE USER pss;
+CREATE DATABASE pss WITH OWNER pss;
```
Comment (dawidskainos) [inline db/setup-db.sql:2]: DB name - maybe patient_switching?

**Review by dawidskainos — CHANGES_REQUESTED** [review]

**Review by dawidskainos — CHANGES_REQUESTED** [review]

**Review by bbell-kainos — APPROVED** [review]

**Pull request merged**